client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx

```diff
@@ -205,6 +208,12 @@ const StatsPersonalPurchase = ( {
 	const handlePlanSwap = ( e: React.MouseEvent ) => {
 		e.preventDefault();
 		recordTracksEvent( `calypso_stats_plan_switched_from_personal_to_commercial` );
+
+		// Check for productType to determine navigation
+		if ( productType ) {
```

Comment:
`productType` doesn't seem to be used anywhere, so I guess we could just remove it maybe? Or you have other considerations here?

Reply:
My consideration here is with the old wizard. If productType is in the URL, then it's using the new single item purchase screen and we want it to use the URL. and if productType is not in the URL, then it's still using the old wizard and we want it to continue with the old logic (for now, until we discontinue) - but would love @grzegorz-cp input here too just in case 😄

Reply:
Good thinking about finding a way to distinguish the flows. 
The downside of checking for the parameter is that if someone uses the new single-purchase page for the first time, they won't have the URL parameter (it's added only when the upgrade button redirects users).
Because `stats-purchase-single-item.tsx` is used only with the new flow we can pass different `handlePlanSwap` implementations to `<PersonalPurchase />`.

This way `stats-purchase-wizard` passes one `handlePlanSwap` and `stats-purchase-single-item` passes another.

The conclusion is that the update should work even if we remove the check for the parameter.

Reply:
Ah, gotcha. Thanks for clarifying that! Okay let's pull out the parameter check and see how it goes 😄